### PR TITLE
Fix symlinks when `CMAKE_INSTALL_PREFIX` is `/` or empty

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
Fixes https://github.com/rui314/mold/pull/846#issuecomment-1318000326 reported by @hwti (sorry, hope this is the last issue). This should only affect the cases where `CMAKE_INSTALL_PREFIX` seen by `mold_install_relative_symlink` is empty.

```
root@15bdffb63b4f:~/mold/build# cmake .. -DCMAKE_INSTALL_PREFIX=""

root@15bdffb63b4f:~/mold/build# cmake --install .
-- Install configuration: "Release"
-- Up-to-date: /lib/mold/mold-wrapper.so
-- Up-to-date: /bin/mold
-- Up-to-date: /share/doc/mold/LICENSE
-- Up-to-date: /share/man/man1/mold.1
-- Installing symlink: //libexec/mold/ld -> ../../bin/mold
-- Installing symlink: //bin/ld.mold -> mold
-- Installing symlink: //bin/ld64.mold -> mold
-- Installing symlink: //share/man/man1/ld.mold.1 -> mold.1

root@15bdffb63b4f:~/mold/build# cmake --install . --prefix /usr/local
-- Install configuration: "Release"
-- Up-to-date: /usr/local/lib/mold/mold-wrapper.so
-- Up-to-date: /usr/local/bin/mold
-- Up-to-date: /usr/local/share/doc/mold/LICENSE
-- Up-to-date: /usr/local/share/man/man1/mold.1
-- Installing symlink: //usr/local/libexec/mold/ld -> ../../bin/mold
-- Installing symlink: //usr/local/bin/ld.mold -> mold
-- Installing symlink: //usr/local/bin/ld64.mold -> mold
-- Installing symlink: //usr/local/share/man/man1/ld.mold.1 -> mold.1

root@15bdffb63b4f:~/mold/build# cmake --install . --prefix .
-- Install configuration: "Release"
-- Up-to-date: /root/mold/build/./lib/mold/mold-wrapper.so
-- Up-to-date: /root/mold/build/./bin/mold
-- Up-to-date: /root/mold/build/./share/doc/mold/LICENSE
-- Up-to-date: /root/mold/build/./share/man/man1/mold.1
-- Installing symlink: //root/mold/build/libexec/mold/ld -> ../../bin/mold
-- Installing symlink: //root/mold/build/bin/ld.mold -> mold
-- Installing symlink: //root/mold/build/bin/ld64.mold -> mold
-- Installing symlink: //root/mold/build/share/man/man1/ld.mold.1 -> mold.1
```

In addition, I added an additional commit to tweak the "Installing symlink" output. That extra slash (as seen in the output above) can be safely removed because `NEW_ABS` is absolute and therefore always begins with a slash.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -388,12 +388,16 @@ if(NOT CMAKE_SKIP_INSTALL_RULES)
 
   function(mold_install_relative_symlink OLD NEW)
     install(CODE "
-      get_filename_component(PREFIX_ABS \${CMAKE_INSTALL_PREFIX} ABSOLUTE)
+      if(\"\${CMAKE_INSTALL_PREFIX}\" STREQUAL \"\")
+        set(PREFIX_ABS \"/\")
+      else()
+        get_filename_component(PREFIX_ABS \${CMAKE_INSTALL_PREFIX} ABSOLUTE)
+      endif()
```
Comment (rui314) [inline CMakeLists.txt:395]: I wonder if

```
get_filename_component(PREFIX_ABS \${CMAKE_INSTALL_PREFIX}/ ABSOLUTE)
```

works instead (notice the trailing `/` at the end of `CMAKE_INSTALL_PREFIX`).

Comment (ZhongRuoyu) [reply CMakeLists.txt]: Thanks! Indeed it's working, and simpler.

**Comment by hwti** [comment]
This works fine for me.

But the commit message is incorrect, empty and / are different.

`-DCMAKE_INSTALL_PREFIX=` :
```
-- Install configuration: "Release"
-- Installing: /lib/mold/mold-wrapper.so
-- Installing: /bin/mold
-- Installing: /share/doc/mold/LICENSE
-- Installing: /share/man/man1/mold.1
-- Installing symlink: /libexec/mold/ld -> ../../bin/mold
-- Installing symlink: /bin/ld.mold -> mold
-- Installing symlink: /bin/ld64.mold -> mold
-- Installing symlink: /share/man/man1/ld.mold.1 -> mold.1
```

`-DCMAKE_INSTALL_PREFIX=/` :
```
-- Install configuration: "Release"
-- Installing: /usr/lib/mold/mold-wrapper.so
-- Installing: /usr/bin/mold
-- Installing: /usr/share/doc/mold/LICENSE
-- Installing: /usr/share/man/man1/mold.1
-- Installing symlink: /usr/libexec/mold/ld -> ../../bin/mold
-- Installing symlink: /usr/bin/ld.mold -> mold
-- Installing symlink: /usr/bin/ld64.mold -> mold
-- Installing symlink: /usr/share/man/man1/ld.mold.1 -> mold.1
```

**Comment by ZhongRuoyu** [comment]
> But the commit message is incorrect, empty and / are different.

Thanks for pointing that out. I've made the correction.

**Pull request merged**